ruby/tests/common_tests.rb

```diff
@@ -870,6 +870,9 @@ def test_protobuf_encode_decode_json_helpers
 
     decoded_msg = Google::Protobuf.decode_json(proto_module::TestMessage, encoded_msg)
     assert_equal proto_module::TestMessage.decode_json(m.to_json), decoded_msg
+
+    assert_equal [m].to_json, Google::Protobuf.encode_json([m])
```

Comment:
Maybe I am missing something, but this test doesn't seem to exercise the new code. We need a test that calls `encode_json` with a second parameter that response to `to_h` but is not itself a `Hash`.